### ISSUE TITLE
[Parser] `Tokenizer Error` on list comprehension example

### ISSUE THREAD

**Issue by akshanshbhatt** [lcompilers/lpython#752]
This example seems to fail on the new parser -

```py
❯ cat examples/expr2.py
[i for (i) in range(5)]
❯ lpyast examples/expr2.py
(Module [(Expr (ListComp (Name i Load) [((Name i Store) (Call (Name range Load) [(ConstantInt 5 ())] []) [] 0)]))] [])
❯ lpyast examples/expr2.py --new-parser
tokenizer error: Parentheses does not match
 --> examples/expr2.py:1:10
  |
1 | [i for (i) in range(5)]
  |          ^


Note: if any of the above error or warning messages are not clear or are lacking
context please report it to us (we consider that a bug that needs to be fixed).
```

However, the tokenizer works as expected and doesn't throw any error message when tested independently. Tokenizer's output -

```py
❯ lpy --show-tokens examples/expr2.py
(TOKEN "[") 0:0
(TOKEN "identifier" i) 1:1
(KEYWORD "for") 3:5
(TOKEN "(") 7:7
(TOKEN "identifier" i) 8:8
(TOKEN ")") 9:9
(KEYWORD "in") 11:12
(TOKEN "identifier" range) 14:18
(TOKEN "(") 19:19
(TOKEN "integer" 5) 20:20
(TOKEN ")") 21:21
(TOKEN "]") 22:22
(NEWLINE) 23:23
(EOF) 24:24
```

The part of the code that throws the exception also runs fine when tested individually -

```py
❯ cat examples/expr2.py
(i)
❯ lpyast examples/expr2.py --new-parser
(Module [(Expr (Name i Load))] [])
❯ lpyast examples/expr2.py
(Module [(Expr (Name i Load))] [])
```

**Comment by certik** [comment]
It's probably a bug in parentheses matching in the tokenizer.
